company-ycmd.el

```diff
@@ -519,6 +526,8 @@ If CB is non-nil, call it with candidates."
                    (get-text-property 0 'params candidate))
     (when (memq major-mode '(python-mode rust-mode))
       (setq it (company-ycmd--remove-self-from-function-args it)))
+    (when (eq major-mode 'rust-mode)
```

Comment:
Phew, I didn't think this function would be that performance critical :smile: . Changed.